Sources/NIO/GetaddrinfoResolver.swift

```diff
@@ -102,13 +141,44 @@ internal class GetaddrinfoResolver: Resolver {
             /* this is odd, getaddrinfo returned NULL */
             self.fail(SocketAddressError.unsupported)
         }
+#endif
     }
 
     /// Parses the DNS results from the `addrinfo` linked list.
     ///
     /// - parameters:
     ///     - info: The pointer to the first of the `addrinfo` structures in the list.
     ///     - host: The hostname we resolved.
+#if os(Windows)
+    private func parseResults(_ pResult: UnsafeMutablePointer<ADDRINFOW>, host: String) {
+        var v4: [SocketAddress] = []
+        var v6: [SocketAddress] = []
+
+        var pResult: UnsafeMutablePointer<ADDRINFOW>? = pResult
+        while pResult != nil {
```

Comment:
We save ourselves some force unwraps by using `while let` instead of `while != nil`. Even better we could follow the looping logic below which uses `while true` and explicitly breaks the loop if it cannot assign a non-nil pointer.

Reply:
Thanks for pointing that out.  No offense, but I kinda disagree, after looking at it when you pointed it out, I think that this entire change is just rubbish.  There is no need for the difference here - this is a private function, an internal typealias is the right thing to do here.  No one cares about internal typealiases - its the public surface that matters.  By doing this this change becomes a mere 5 lines.

Sorry.

Reply:
Do we want to do the same for the other lookup function here?

Reply:
Discussed offline, although for the core body we could use a `let` binding to deal with the differences in functions, the string encoding I feel like would detract from readability as the string re-encoding can be confusing.